src/transformStateWithClones.js

```diff
@@ -6,8 +6,35 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const res = [];
+  const clone = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        Object.assign(clone, actions[i].extraData);
+
+        break;
+
+      case 'removeProperties':
+        actions[i].keysToRemove.forEach((x) => {
+          delete clone[`${x}`];
+        });
+        break;
+
+      case 'clear':
+        for (const key in clone) {
+          delete clone[key];
+        }
+        break;
```

Comment:
![Screenshot 2022-08-29 at 09 00 46](https://user-images.githubusercontent.com/99331793/187133018-23423edb-86f6-470b-80dd-b0731cae8eb4.png)

Reply:
I don't understand how I should apply destructuring here and what exactly I should specify in default.
Help me please

Reply:
`const {type, extraData, keysToRemove } = action`

Reply:
`default: 
 break;
`